File: src/models/Product.ts
import {
  Entity,
  Column,
  PrimaryGeneratedColumn,
  ManyToOne,
  JoinColumn,
} from 'typeorm';
import Category from './Category';

@Entity('products')
class Product {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column()
  name: string;

  @Column({ nullable: true })
  description: string;

  @Column({ nullable: true })
  price: number;

  @Column({ nullable: true })
  promo_price: number;

  @Column({ nullable: true })
  status_flag: string;

  @Column({ nullable: true })
  image_url: string;

  @ManyToOne(() => Category)
  @JoinColumn({ name: 'category_id' })
  category: Category;
}

export default Product;

File: lib/models/product.dart
import 'package:flutter/foundation.dart';

class Product {
  final int id;
  final String name;
  final String description;
  final String price;
  final String promotionalPrice;
  final String statusFlag;
  final String category;

  Product(
      {@required this.id,
      @required this.name,
      this.description,
      this.price,
      this.promotionalPrice,
      this.statusFlag,
      this.category});

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
        id: json['id'] as int,
        name: json['name'] as String,
        description: json['description'] as String,
        price: json['price'] as String,
        promotionalPrice: json['promotional_price'] as String,
        statusFlag: json['status_flag'] as String,
        category: json['category'] as String);
  }
}
